.github/workflows/code-audit.yaml

```diff
@@ -0,0 +1,38 @@
+name: Code audit
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  lint-build-test:
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - uses: pnpm/action-setup@v2
+      with:
+        version: 8
+    - name: Install Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: 20.x
+        cache: 'pnpm'
+    - name: Install dependencies
+      run: pnpm run verify-deps
+    - name: Lint code
+      run: pnpm run lint
+    - name: Test smoke
+      run: pnpm run build
+    - name: Test unit
+      run: pnpm run test:unit
+    - name: Test E2E (Linux)
+      if: runner.os == 'Linux'
+      run: xvfb-run -a pnpm run test:e2e
+    - name: Test E2E (non-Linux)
+      if: runner.os != 'Linux'
+      run: pnpm run test:e2e
```

Comment:
so, these don't sound like `code audit` to me... 🤔 

would you prefer to split into a new workflow or rename this workflow ?

Reply:
For now I've kept it simple and dumped it all in a single job of a single workflow. I can definitely be optimized, parallelized further etc, but I opted to not spend time on that until there's need for it. Current pipeline takes 30"-2' minutes, which I called good enough.

Another reason why I have it like this is because I want all those audits (lint, build, test unit, test e2e) to all run an each of the three platforms (linux, windows, macos). The environment can affect everything. For example the linting step was failing only on windows for a very obscure reason (see commit for more details).
So this collection of steps runs x3 (once for each platform). If we split in two workflows we'll have 6 runs. And we already have 3 more (commit lint, pr lint, codeql).
That's what I meant by keeping it simple and pragmatic.

In my head code audit is a matching name, it audits that the code complies with and doesn't break any tests.

So choose an option to proceed:

1. split into multiple jobs / workflows
2. rename existing workflow and or job
3. something else?
4. resolve discussion

Pick one.

Reply:
2. rename to `ci.yaml`